File: code/pascal/control-flow/TestBools.pas
program TestBools;

procedure PrintBoolean(value: Boolean);
begin
    if value then   // if value is true then...
        Write('true')
    else 
        Write('false');
end;

// Is v1 at least double v2
function AtLeastDouble(v1, v2: Integer): Boolean;
begin
    result := v1 >= 2 * v2;
end;

procedure Main();
var
  test: Boolean;
  num: Integer;
begin
    Write('Enter a number: ');
    ReadLn(num);
    
    test := AtLeastDouble(num, 5);
    Write(num, ' is at least double 5 is ');
    PrintBoolean(test);
    WriteLn();
end;

begin
  Main();
end.

File: code/c/control-flow/test-bools.c
/* Program: test-bools.c */
#include <stdio.h>
#include <stdbool.h>

void print_bool(bool value)
{
    if(value) 
        printf("true");
    else 
        printf("false");
}

// Is v1 at least double v2
bool at_least_double(int v1, int v2)
{
    return v1 >= 2 * v2;
}

int main()
{
    bool test;
    int num;
    
    printf("Enter a number: ");
    scanf("%d", &num);
    
    test = at_least_double(num, 5);
    printf("%d is at least double 5 is ", num);
    print_bool(test);
    printf("\n");
    
    return 0;
}